modules/integration/src/test/scala/zio/elasticsearch/HttpExecutorSpec.scala

```diff
@@ -2771,6 +2772,69 @@ object HttpExecutorSpec extends IntegrationSpec {
             }
           } @@ after(Executor.execute(ElasticRequest.deleteIndex(geoPolygonIndex)).orDie)
         ),
+        suite("intervals query")(
+          test("intervalMatch returns only matching document") {
+            checkOnce(genDocumentId, genTestDocument, Gen.alphaNumericString, genDocumentId, genTestDocument) {
+              (idMatch, docMatch, targetWord, idNoMatch, docNoMatch) =>
+                val docShouldMatch    = docMatch.copy(stringField = s"prefix $targetWord suffix")
+                val docShouldNotMatch = docNoMatch.copy(stringField = "completely unrelated text")
+                val field             = TestDocument.stringField.toString
+                val query             = intervals(field, intervalMatch(targetWord))
+
+                for {
+                  _ <- Executor.execute(ElasticRequest.deleteByQuery(firstSearchIndex, matchAll))
+                  _ <- Executor.execute(ElasticRequest.upsert(firstSearchIndex, idMatch, docShouldMatch))
+                  _ <- Executor.execute(
+                         ElasticRequest.upsert(firstSearchIndex, idNoMatch, docShouldNotMatch).refreshTrue
+                       )
+                  res <- Executor.execute(ElasticRequest.search(firstSearchIndex, query)).documentAs[TestDocument]
+                } yield assert(res)(
+                  Assertion.contains(docShouldMatch) &&
+                    Assertion.not(Assertion.contains(docShouldNotMatch))
```

Comment:
No need for `docShouldNotMatch`. Maybe you can use `containsOnly` or something like that.